prow/sidecar/run.go

```diff
@@ -125,7 +129,12 @@ func (o Options) Run(ctx context.Context) (int, error) {
 				// second upload but we can tolerate this as we'd rather get SOME
 				// data into GCS than attempt to cancel these uploads and get none.
 				logrus.Errorf("Received an interrupt: %s, cancelling...", s)
-				cancel()
+
+				err = o.doUpload(spec, false, true, metadata, buildLogs)
+				if err != nil {
+					logrus.Errorf("Failed to perform best-effort upload: %v", err)
+				}
```

Comment:
Please also log an `Info` level log if the best-effort upload is successful. That will make it easier to determine if the best-effort upload overlaps with the main upload temporally while debugging.

Reply:
Added this inside `doBestEffortUpload` method.